game/gamestates/start_menu.lua

```diff
@@ -46,18 +49,38 @@ end
 
 function state:update (dt)
   INPUT.update()
-  if MENU.begin("START_MENU", 80*4, _height / 2) then
-    if MENU.item("New route") then
-      SWITCHER.switch(GS.PLAY)
+  if MENU.begin(_menu_context, 80*4, _height / 2, 3, 160) then
+    if _menu_context == "START_MENU" then
+      if MENU.item("New route") then
+        local route_id = PROFILE.newRoute()
+        SWITCHER.switch(GS.PLAY, route_id)
+      end
+      if MENU.item("Load route") then
+        _menu_context = "LOAD_LIST"
+      end
+      if MENU.item("Quit") then
+        love.event.quit()
+      end
+    elseif _menu_context == "LOAD_LIST" then
+      local savelist = PROFILE.getSaveList()
+      if #savelist > 0 then
+        for _,route_id in ipairs(savelist) do
+          if MENU.item(route_id) then
+            SWITCHER.switch(GS.PLAY, PROFILE.loadRoute(route_id))
```

Comment:
On line 56 you passed the route ID to the PLAY gamestate, but here you pass the actual route data. Which one is the intended usage?

Reply:
I have no idea. We need to determine what builds the save. It might be the route, the profile, or it might be something else entirely. I guess the `PROFILE` can fetch the route data from the savefile, but can it, or should it, generate route data when creating a new route?

Reply:
I guess it's probably better to have a separate class `RouteGenerator` or equivalent (Maker? Builder?). `PROFILE` uses requests fresh new route data from it and passes to `PLAY` gamestate. Something like that?

Reply:
Though this is likely not necessary for this PR.

Reply:
That sounds good, I think. For now I guess it can be a simple table generator? I will can add it to the PR

Reply:
OK